internal/spoke_k8s_client/factory_test.go

```diff
@@ -0,0 +1,66 @@
+package spoke_k8s_client
```

Comment:
Do we have any tests that validate that a valid kubeconfig is produced at the end of this process?